### PR TITLE
Add gscam 1.0.1 to kinetic (taken from melodic)

### PR DESCRIPTION
**Pull request by awesomebytes**

### CODE REVIEW

**Comment by awesomebytes** [comment]
Sorry about the multiple commits. Maybe you can squash them together with the Github interface on merge?

Also, this is 'necessary' cause gscam pulls gstreamer-1.0, and, at least in my system, there was also gstreamer-0.10 (pulled by some previous dependency I guess). The CMakeLists.txt checks first for gstreamer-0.10, so in my case it failed when executing gscam as v4l2src was not present (on 0.10). As we are on Gentoo and we tend to go with the bleeding edge, I think it makes sense to go with Gstreamer 1.0 (and so with gscam 1.0.1 forcing to use Gstreamer 1.0).

**Comment by awesomebytes** [comment]
Hmm I see what happened. Gstreamer 0.10 was pulled as a dependency when I installed ros-kinetic/gscam.

Then it complained cause it couldn't find v4l2src as it was missing media-libs/gst-plugins-good, so I emerged that... BUT, it got version 1.14.4, so gscam was still trying to call gstreamer 0.10 v4l2src... and that didn't work. Then I got confused and I just went on the way of using a newer version of gscam with support for gstreamer 1.0... and I ended up with this PR.

I'd prefer to have a newer version of gscam with Gstreamer 1.0 to be honest.

I asked in https://github.com/ros-drivers/gscam/issues/25#issuecomment-454428680 if it would be possible to release on kinetic that version 1.0.1, but I'm guessing that if they didn't is cause there is some issue.

File: ros-kinetic/gscam/gscam-1.0.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+EAPI=6
+PYTHON_COMPAT=( python{2_7,3_5,3_6} )
+
+inherit ros-cmake
+
+DESCRIPTION="A ROS camera driver that uses gstreamer to connect to\
+	devices such as [...]"
+HOMEPAGE="https://wiki.ros.org"
+SRC_URI="https://github.com/ros-drivers-gbp/${PN}-release/archive/release/melodic/${PN}/1.0.1-0.tar.gz -> ${PN}-melodic-release-${PV}.tar.gz"
```
Comment (awesomebytes) [inline ros-kinetic/gscam/gscam-1.0.1.ebuild]: @allenh1 I think this is why you see a mention of melodic. It should be harmless, as it's just a name of the file.

**Comment by allenh1** [comment]
I really don't know about this. I am more in favor of waiting for kinetic to get a release bump than using the melodic build here.

@tfoote thoughts?

**Comment by awesomebytes** [comment]
Closing this PR as it was mostly a mess. (and it will live in Google's archive!)
I ended up not using the work... the driver used more CPU than the previous driver I was using.

**Pull request closed without merging**